controllers/managedocs_controller.go

```diff
@@ -1752,6 +1759,41 @@ func (r *ManagedOCSReconciler) updateMCGCSV(csv *opv1a1.ClusterServiceVersion) e
 	return nil
 }
 
+func (r *ManagedOCSReconciler) reconcileConsolePlugin() error {
+	consolePlugin := &consolev1alpha1.ConsolePlugin{}
+	consolePlugin.Name = consolePluginName
+	consolePlugin.Namespace = r.namespace
+	err := r.get(consolePlugin)
+	if err != nil {
+		// ODF Operator has to bring this CR,
+		// so if not found, we silently return
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	expectedProxy := consolev1alpha1.ConsolePluginProxy{
+		Type:      consolev1alpha1.ProxyTypeService,
+		Authorize: true,
+		Alias:     consoleProxyAlias,
+		Service: consolev1alpha1.ConsolePluginProxyServiceConfig{
+			Name:      prometheusServiceName,
+			Namespace: r.namespace,
+			Port:      int32(templates.KubeRBACProxyPortNumber),
+		},
+	}
+	// if any of the ConsolePluginProxy matches the expected proxy,
+	// return happily / without-any-error
+	for _, consoleProxy := range consolePlugin.Spec.Proxy {
```

Comment:
This for-range will create an in-memory copy of each proxy in the `consolePlugin.Spec.Proxy` (as the the array is not a pointer array).

Please use an index and a pointer instead
```suggestion
	for i := range consolePlugin.Spec.Proxy {
		consoleProxy := &consolePlugin.Spec.Proxy[i]
```